File: services/tradelog/tradelog-api-spec/src/main/java/com/example/tradelog/api/spec/model/OptionType.java
package com.example.tradelog.api.spec.model;

public enum OptionType {
    UNKNOWN,
    PUT,
    CALL;

    static public OptionType lookup(String id) {
        try {
            return OptionType.valueOf(id);
        } catch (IllegalArgumentException ex) {
            return UNKNOWN;
        }
    }
}

File: services/tradelog/tradelog-api/src/main/kotlin/com/example/tradelog/api/core/model/OptionType.kt
package com.example.tradelog.api.core.model

enum class OptionType {
    UNKNOWN,
    PUT,
    CALL;

    companion object {
        fun lookup(id: String) : OptionType {
            return try {
                valueOf(id)
            } catch (ex: IllegalArgumentException) {
                UNKNOWN
            }
        }
    }
}
